Encryption.md

```diff
@@ -186,47 +372,77 @@ columns in encrypted files - at a price of leaving certain metadata fields unpro
 (not encrypted or tamper-proofed). The plaintext footer mode can be useful during a transitional period 
 in organizations 
 where some frameworks can't be upgraded to a new Parquet library for a while. Data writers will
-upgrade and run with a new Parquet version, producing encrypted files in this mode. Data readers, 
-working with a sensitive data, will also upgrade to a new Parquet library. But other readers that
+upgrade and run with a new Parquet version, producing encrypted files in this mode. Data readers 
+working with sensitive data will also upgrade to a new Parquet library. But other readers that
 don't need the sensitive columns, can continue working with an older Parquet version. They will be 
 able to access plaintext columns in encrypted files. A legacy reader, trying to access a sensitive 
-column data in a ".parquet.encrypted" file with a plaintext footer, will get an  exception. More
+column data in an encrypted file with a plaintext footer, will get an  exception. More
 specifically, a Thrift parsing exception on an encrypted `PageHeader` structure. Again, using legacy
 Parquet readers for encrypted files is a temporary solution.
 
 In the plaintext footer mode, the `optional ColumnMetaData meta_data` is set in the `ColumnChunk` 
 structure for all columns, but is stripped of the statistics for the sensitive (encrypted) columns. 
-These statistics are available for new readers with the column key - they fetch the split ColumnMetaData, 
-and decrypt it to get all metadata values. The legacy readers are not aware of the split metadata, 
+These statistics are available for new readers with the column key - they fetch the separated ColumnMetaData, 
+and decrypt it to get all metadata values. The legacy readers are not aware of the separated metadata; 
 they parse the embedded field as usual. While they can't read the data of the encrypted columns, they 
-read the metadata to exract the offset and size of the column data - required for input vectorization
+read the metadata to extract the offset and size of the column data - required for input vectorization
 (see the next section).
 
 An `encryption_algorithm` field is set at the FileMetaData structure. Then the footer is written as usual, 
 followed by its length (4-byte little endian integer) and a final magic string, "PAR1".
+The same magic string is written at the beginning of the file (offset 0). 
+The magic bytes for plaintext footer mode are ‘PAR1’ to allow older readers to read projections of the 
+file that do not include encrypted columns.
+
+```c
+struct FileMetaData {
+...
+  /** 
+   * Encryption algorithm. Note that this field is only used for files
+   * with plaintext footer. Files with encrypted footer store the algorithm id
+   * in FileCryptoMetaData structure.
+   */
+  8: optional EncryptionAlgorithm encryption_algorithm
```

Comment:
Is this field set when using an encrypted footer? Its value should always be specified.

Reply:
This field is only used for files with plaintext footer. It is not set for files with encrypted footers. I'll add this to the comment.